DB/src/main/java/io/deephaven/db/v2/locations/local/ReadOnlyParquetTableLocation.java

```diff
@@ -47,10 +48,16 @@
 
     private final Map<String, GroupingFile> groupingFiles = new ConcurrentHashMap<>();
     private final Set<String> grouping = new HashSet<>();
-
-    ReadOnlyParquetTableLocation(@NotNull TableKey tableKey, @NotNull TableLocationKey tableLocationKey, File parquetFile, boolean supportsSubscriptions) {
+    private final Map<String, String> columnNameMapping;
+
+    ReadOnlyParquetTableLocation(
+            @NotNull final TableKey tableKey,
+            @NotNull final TableLocationKey tableLocationKey,
+            final File parquetFile,
+            final boolean supportsSubscriptions,
+            final Map<String, String> columnNameMapping) {
```

Comment:
I think you could make this code better and more flexible by taking a `java.util.function.Function<String, String>` or a similar-but-better-documented functional interface and using that as the mapper, rather than an actual explicit `Map<String, String>`.

Reply:
More discussion in answer to later comments.